src/types/inc/IInputEvent.hpp

```diff
@@ -514,12 +514,14 @@ class FocusEvent : public IInputEvent
 {
 public:
     constexpr FocusEvent(const FOCUS_EVENT_RECORD& record) :
-        _focus{ !!record.bSetFocus }

```

Comment:
and we're certain that nobody else uses this ctor?

Reply:
Other than `WriteConsoleInputAImpl` and `WriteConsoleInputWImpl`, it's potentially used in a few places in the `InputBuffer` class when moving `IInputEvent` records around (calling `IInputEvent::Create` on the result of a `ToInputRecord` call). I don't think those are likely to be a problem though.

`IInputEvent::Create` is also called in a couple of places in `InputStateMachineEngine`, but those are exclusively for keyboard and mouse events as far as I can see.

I think the rest are all in tests, which I didn't bother looking at, but I see now they've resulted in some test failures. Hopefully those are just tests that need to be updated, though, and not an indication that there's something fundamentally wrong with this approach.